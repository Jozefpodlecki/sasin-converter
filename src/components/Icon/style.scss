.icon {
    transition: opacity .2s ease-in;
    cursor: pointer;
    user-select: none;
    
    &:hover {
        opacity: 0.5;
    }
}