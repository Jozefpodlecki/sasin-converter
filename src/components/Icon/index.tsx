import { IconDefinition } from "@fortawesome/free-solid-svg-icons";
import React, {FunctionComponent } from "react";
import { __RouterContext } from "react-router";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

import style from "./style.scss";
import { SizeProp } from "@fortawesome/fontawesome-svg-core";

type Prop = {
    className?: string;
    icon: IconDefinition;
    size?: SizeProp;
    color?: string;
}

const defaultColor = "white";

const Icon: FunctionComponent<Prop> = ({
    className,
    icon,
    size,
    color
}) => {
	return <div className={`${className || ""} ${style.icon}`}>
        <FontAwesomeIcon icon={icon} color={color || defaultColor} size={size} />
    </div>
  };
  
  export default Icon;