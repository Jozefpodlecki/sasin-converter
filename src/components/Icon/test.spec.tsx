import React from "react"
import { render, fireEvent } from "@testing-library/react"
import Icon from "./index"
import { faGem } from "@fortawesome/free-solid-svg-icons";

describe('Icon spec', () => {
    
    it('renders the component', () => {
        const container = render(<Icon icon={faGem} />)
        expect(container).toMatchSnapshot();
    });
   
});