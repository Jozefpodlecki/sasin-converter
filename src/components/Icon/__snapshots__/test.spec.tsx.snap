// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon spec renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=" icon"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-gem fa-w-18 "
          color="white"
          data-icon="gem"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M485.5 0L576 160H474.9L405.7 0h79.8zm-128 0l69.2 160H149.3L218.5 0h139zm-267 0h79.8l-69.2 160H0L90.5 0zM0 192h100.7l123 251.7c1.5 3.1-2.7 5.9-5 3.3L0 192zm148.2 0h279.6l-137 318.2c-1 2.4-4.5 2.4-5.5 0L148.2 192zm204.1 251.7l123-251.7H576L357.3 446.9c-2.3 2.7-6.5-.1-5-3.2z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=" icon"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-gem fa-w-18 "
        color="white"
        data-icon="gem"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M485.5 0L576 160H474.9L405.7 0h79.8zm-128 0l69.2 160H149.3L218.5 0h139zm-267 0h79.8l-69.2 160H0L90.5 0zM0 192h100.7l123 251.7c1.5 3.1-2.7 5.9-5 3.3L0 192zm148.2 0h279.6l-137 318.2c-1 2.4-4.5 2.4-5.5 0L148.2 192zm204.1 251.7l123-251.7H576L357.3 446.9c-2.3 2.7-6.5-.1-5-3.2z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
